code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm

```diff
@@ -37,161 +41,370 @@ Doesn't work on other aliens/AI.*/
+		return 0
+	return ..()
+
+/spell/aoe_turf/evolve/drone/spell_do_after(var/mob/user as mob, delay as num, var/numticks = 5)
+	user.visible_message("<span class='alien'>[src] begins to violently twist and contort!</span>", "<span class='alien'>You begin to evolve, stand still for a few moments</span>")
+	return ..()
+
+/spell/aoe_turf/evolve/drone/cast(list/targets, mob/living/carbon/user)
+	..()
+	var/mob/living/carbon/alien/humanoid/queen/new_xeno = new(get_turf(user))
+	for(var/datum/language/L in user.languages)
+		new_xeno.add_language(L.name)
+	user.mind.transfer_to(new_xeno)
+	user.transferImplantsTo(new_xeno)
+	user.transferBorers(new_xeno)
+	qdel(user)
+
+////////////////////////////
+//// FOR THE LARVA BROS ////
+////////////////////////////
+
+/spell/aoe_turf/evolve/larva
+	desc = "Evolve into a fully grown Alien."
+	insufficient_holder_msg = "<span class='alien'>You are not fully grown yet.</span>"
+
+	holder_var_type = "growth"
+	holder_var_amount = LARVA_GROW_TIME
+
+	var/spawning
+
+/spell/aoe_turf/evolve/larva/spell_do_after(mob/user)
+	var/explanation_message = {"<span class='notice'><B>You are growing into a beautiful alien! It is time to choose a caste.</B><br>
+	There are three castes to choose from:<br>
+	<B>Hunters</B> are strong and agile, able to hunt away from the hive and rapidly move through ventilation shafts. Hunters generate plasma slowly and have low reserves.<br>
+	<B>Sentinels</B> are tasked with protecting the hive and are deadly up close and at a range. They are not as physically imposing nor fast as the hunters.<br>
+	<B>Drones</B> are the working class, offering the largest plasma storage and generation. They are the only caste which may evolve again, turning into the dreaded alien queen."}
+	to_chat(user, explanation_message)
+	spawning = input(user, "Please choose which alien caste you shall evolve to.", "Evolving Choice Menu", null) as null|anything in list("Hunter","Sentinel","Drone","Repeat Explanation")
+	while(spawning == "Repeat Explanation")
+		to_chat(user, explanation_message)
+		spawning = input(user, "Please choose which alien caste you shall evolve to.", "Evolving Choice Menu", null) as null|anything in list("Hunter","Sentinel","Drone","Repeat Explanation")
+	if(spawning == null)
+		return 0
+	switch(spawning)
+		if("Hunter")
+			spawning = /mob/living/carbon/alien/humanoid/hunter
+		if("Sentinel")
+			spawning = /mob/living/carbon/alien/humanoid/sentinel
+		if("Drone")
+			spawning = /mob/living/carbon/alien/humanoid/drone
+	return ..()
+
+/spell/aoe_turf/evolve/larva/cast(list/targets, mob/living/carbon/user)
+	var/mob/living/carbon/alien/humanoid/new_xeno = new spawning(get_turf(user))
+	for(var/datum/language/L in user.languages)
+		new_xeno.add_language(L.name)
+	if(user.mind)
+		user.mind.transfer_to(new_xeno)
+	user.transferImplantsTo(new_xeno)
+	user.transferBorers(new_xeno)
+	qdel(user)
+
+/spell/aoe_turf/alien_hide
```

Comment:
Honestly I don't see much room to make this spell even more general than this, almost everything in it is a type specific override. Either way hide itself is only compatible with mobs who have their reset_plane_and_layer() set up in a specific way to not auto-reset every tick.